src/map/storage.c

```diff
@@ -345,26 +436,35 @@ static int storage_add_to_inventory(struct map_session_data *sd, int index, int
  *   0 : fail
  *   1 : success
  *------------------------------------------*/
-static int storage_storageaddfromcart(struct map_session_data *sd, int index, int amount)
+static int storage_storageaddfromcart(struct map_session_data *sd, struct storage_data* stor, int index, int amount)
 {
 	nullpo_ret(sd);
+	nullpo_ret(stor);
+	Assert_ret(stor->received);
 
-	Assert_ret(sd->storage.received == true);
+	const struct storage_settings* stst = NULL;
+	nullpo_ret(stst = storage->get_settings(stor->uid));
 
-	if (sd->storage.aggregate > MAX_STORAGE)
+	if ((sd->storage.access & STORAGE_ACCESS_PUT) == 0) {
+		clif->delitem(sd, index, amount, DELITEM_NORMAL);
+		clif->additem(sd, index, amount, 0);
+		return 0;
+	}
```

Comment:
I did not test, but another question here: `clif->delitem` and `clif->additem` are related to player inventory, in this case, shouldn't we use the calls for Cart instead (`clif->cart_additem` / `clif->cart_delitem`)? since this is a move from cart to storage

Reply:
Just tested. You're right, should be `clif->cart_delitem` / `clif->cart_additem`. Changed.